hw1-app/src/app/core/auth.service.ts

```diff
@@ -0,0 +1,37 @@
+import {Injectable} from '@angular/core';
+import {Router} from '@angular/router';
+import {Subject} from 'rxjs';
```

Comment:
Minor: this import of rxjs not optimal use rxjs/internal/observable for less bundle